src/core/actuators/engine.hpp

```diff
@@ -12,13 +12,33 @@ inline Engine<Timeseries_t>::Engine(Xml_document& database, const std::string& p
     if ( _only_max_power )
     {
         read_parameters(database, path, get_parameters(), __used_parameters);
+
+        if(database.has_element(path+"gear-ratio"))
+        {
+            database.get_element(path+"gear-ratio").set_attribute("__unused__","false");
```

Comment:
Yup, Our algorithm goes through the xml to check if all the parameters are used or not and the xml has parameters like gear ratio and etc that are only read in the case of not max power. This causes an exception that is not really intended. We are ignoring certain parameters in the xml on purpose.

Reply:
But why is it needed in the case `if ( only_max_power )`? 

I was expecting it only in the `else`

Reply:
This is what I was expecting:

```C++
    if ( _only_max_power )
    {
        read_parameters(database, path, get_parameters(), __used_parameters);
    }
    else
    {
        _gear_ratio = database.get_element(path+"gear-ratio").get_value(double());
        database.get_element(path+"gear-ratio").set_attribute("__unused__","false");

        std::vector<scalar> speed = database.get_element(path+"rpm-data").get_value(std::vector<double>())*RPM;
        database.get_element(path+"rpm-data").set_attribute("__unused__","false");

        std::vector<scalar> p = database.get_element(path+"power-data").get_value(std::vector<double>())*CV;
        database.get_element(path+"power-data").set_attribute("__unused__","false");

        _p = sPolynomial(speed,p,speed.size()-1,true);
    } 
```

Reply:
Hey, this will still throw an exception because, this line only declares `maximum-power`
https://github.com/juanmanzanero/fastest-lap/blob/505fc2c834582e3da2dfa776809243106a28eb53/src/core/actuators/engine.h#L67
Whereas the XML has `gear-ratio`, `rpm-data` and `power-data` which are not read, but exist. This is then interpreted as a not all parameters read case. Which is true but causes an exception to catch during construction:
https://github.com/juanmanzanero/fastest-lap/blob/505fc2c834582e3da2dfa776809243106a28eb53/src/test/vehicles/car_road_cartesian.cpp#L131
Which breaks all the tests. We can catch the exception here but that would create a bug foregoing the original intention to not run if a critical parameter is not read. Hence I added the lines here to mark these parameters are read so as to not cause a false exception to raise. This is a bit of hardcoding, we could make it more versatile, by reading these parameters anyway, but not utilizing them in both cases.

Reply:
Do the following:

have the C++ file like this:

```C++
 if ( _only_max_power )
    {
        read_parameters(database, path, get_parameters(), __used_parameters);
    }
    else
    {
        _gear_ratio = database.get_element(path+"gear-ratio").get_value(double());
        database.get_element(path+"gear-ratio").set_attribute("__unused__","false");

        std::vector<scalar> speed = database.get_element(path+"rpm-data").get_value(std::vector<double>())*RPM;
        database.get_element(path+"rpm-data").set_attribute("__unused__","false");

        std::vector<scalar> p = database.get_element(path+"power-data").get_value(std::vector<double>())*CV;
        database.get_element(path+"power-data").set_attribute("__unused__","false");

        _p = sPolynomial(speed,p,speed.size()-1,true);
    } 
```

and remove these elements (`gear-ratio`, `rpm-data` and `power-data`) from the XML database file. This should only break the actuators test but the rest should be [OK]. If this works, we are good to go. I will take care of the failing actuators test later. It's a very dummy and silly test I added during the development early phases.